### PR TITLE
godoc package documentation

### PR DESCRIPTION
**Pull request by Eknir**

### CODE REVIEW

**Comment by acud** [comment]
@Eknir i've hijacked this task. hope you don't mind

File: cmd/bee/doc.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Copyright 2021 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package main provides the bee binary viper-cobra
+// command definitions.
```
[the remark this answers is not in the document]
Comment (acud) [reply cmd/bee/doc.go:6]: OK. I will remove this file. not sure why we need to document this specific package

File: pkg/node/node.go [hunk 2]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// Package node provides the glue-code needed in order
```
[the remark this answers is not in the document]
Comment (acud) [reply pkg/node/node.go]: OK. Had another try at this. better?

File: pkg/kademlia/kademlia.go [hunk 3]
```diff
@@ -2,6 +2,9 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// Package kademlia provides the implementation for the
+// Swarm forwarding-kademlia. Kademlia makes sure you're
```
Comment (Eknir) [inline pkg/kademlia/kademlia.go:6]: Technically, forwarding Kademlia is implemented by this package, but also by the retrieval protocol. I would say:
`Package Kademlia makes sure that you are connect to the right peers at any point in time, based on Kademlia topology. It forms the basis of Swarms forwarding Kademlia`

File: pkg/netstore/netstore.go [hunk 4]
```diff
@@ -2,6 +2,10 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// Package netstore provides an abstraction layer over the
+// Swarm local storage layer that leverages connectivity
+// with other peers in order to retrieve chunks that cannot
+// be found locally from the network.
```
Comment (Eknir) [inline pkg/netstore/netstore.go:8]:
```suggestion
// with other peers in order to retrieve chunks from the network that cannot
// be found locally.
```

File: pkg/p2p/p2p.go [hunk 5]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// Package p2p provides the p2p abstractions used
```
Comment (Eknir) [inline pkg/p2p/p2p.go:5]:
```suggestion
// Package p2p provides the peer-to-peer abstractions used
```

File: pkg/pingpong/pingpong.go [hunk 6]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// Package pingpong exposes the simple ping-pong protocol
+// which measures RTT with other peers.
```
Comment (Eknir) [inline pkg/pingpong/pingpong.go:6]: Please not abbreviations

File: pkg/pss/pss.go [hunk 7]
```diff
@@ -2,6 +2,11 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// Package pss exposes functionalities needed to communicate
+// with other peers on the network. Pss uses pushsync and
+// pullsync for message delivery and mailboxing. Sending and
```
Comment (Eknir) [inline pkg/pss/pss.go:7]:
```suggestion
// with other peers on the network. Pss uses pushsync and
// pullsync for message delivery and mailboxing. All messages are disguised as content-addressed chunks. Sending and
```

File: pkg/puller/puller.go [hunk 8]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// Package puller provides protocol-orchestrating functionality
+// over the pullsync protocol.
```
Comment (Eknir) [inline pkg/puller/puller.go:6]: In `pusher.go`, you write a similar comment, but add:
` (...) makes sure that chunks meant
// to be distributed over the network are sent used using the
// pushsync protocol.`
Do you think it makes sense to add a similar explainer sentence here?

File: pkg/pullsync/pullsync.go [hunk 9]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// Package pullsync provides the pullsync protocol
+// implementation.
```
Comment (Eknir) [inline pkg/pullsync/pullsync.go:6]: That doesn't say a lot to me. Can you please be more specific on the function of pullsync?

File: pkg/pushsync/pushsync.go [hunk 10]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// Package pushsync provides the pushsync protocol
+// implementation.
```
Comment (Eknir) [inline pkg/pushsync/pushsync.go:6]: that doesn't say a lot to me. Can you please be more specific on the function of pushsync?

File: pkg/retrieval/retrieval.go [hunk 11]
```diff
@@ -2,6 +2,9 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// Package retrieval provides the retrieval protocol
+// implementation. The protocol is used to retrieve
+// chunks over the network.
```
Comment (Eknir) [inline pkg/retrieval/retrieval.go:7]: (...) using forwarding-Kademlia routing

File: pkg/soc/soc.go [hunk 12]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// Package soc provides the single-owner chunk implemenation
+// and validator.
```
Comment (Eknir) [inline pkg/soc/soc.go:6]: suggestion to add: An soc is a chunks whose address is derived of (...)

File: pkg/statestore/doc.go [hunk 13]
```diff
@@ -0,0 +1,7 @@
+// Copyright 2021 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package statestore provides statestore leveldb implementation
+// and mock.
```
Comment (Eknir) [inline pkg/statestore/doc.go:6]:
```suggestion
// Package statestore provides statestore implementation
// and mock. It uses leveldb under the hood.
```

**Review by Eknir — COMMENTED** [review]
Great work!

I could not have done this in the way you pulled it of. Thanks for taking this one from me.

I left comments on some of your changes.

One thing I also found that the package documentation of  `swarm/swarm.go` (// Package swarm contains most basic and general Swarm concepts.) should be updated as well. This one doesn't say me anything.

File: pkg/soc/soc.go (hunk 12 above)
Comment (acud) [reply pkg/soc/soc.go:6]: prefer not to get technical in the godoc

File: pkg/pushsync/pushsync.go (hunk 10 above)
Comment (acud) [reply pkg/pushsync/pushsync.go:6]: this is a developer guide, i don't want to document what the protocol does in this description

File: pkg/pullsync/pullsync.go (hunk 9 above)
Comment (acud) [reply pkg/pullsync/pullsync.go:6]: see below

File: pkg/bitvector/bitvector.go [hunk 14]
```diff
@@ -14,6 +14,8 @@
 // You should have received a copy of the GNU Lesser General Public License
 // along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
 
+// Package bitvector provides functionality of a
+// simple bit vector implementation.
```
[the remark this answers is not in the document]
Comment (acud) [reply pkg/bitvector/bitvector.go:18]: we _do not_ update copyright in files

File: pkg/intervalstore/intervals.go [hunk 15]
```diff
@@ -14,6 +14,12 @@
 // You should have received a copy of the GNU Lesser General Public License
 // along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
 
+// Package intervalstore provides a persistence layer
+// for intervals relating to a peer. Intervals are
+// assumed to be integers of type uint64. The store
```
[the remark this answers is not in the document]
Comment (acud) [reply pkg/intervalstore/intervals.go:19]: it is a general component. intervals are intervals (as the name might suggest). a quick look in the code and tests would explain it very quickly even for the non-coder. i will remove the part about uint64

File: pkg/kademlia/kademlia.go (hunk 3 above)
Comment (acud) [reply pkg/kademlia/kademlia.go:6]: I realised that kademlia package already had a description. removing this...

**Review by acud — APPROVED** [review]
@Eknir can't approve since he opened this PR, but told me to give his blessing through my github handle

**Review by janos — APPROVED** [review]
Thanks @acud! 👍

**Pull request merged**